### DISCUSSION TITLE
UGS 2.0.17 crashes on OSX 10.14.6

### DISCUSSION THREAD

**Discussion by dkag123** [winder/Universal-G-Code-Sender#2233] [Help]
Can not open the app, it crashes upon opening

I have also uninstalled all previous versions and deleted the leftover files in Library/ pref/ugsplatform
Still no luck

Has anyone been able to open this on OSX?

**Comment by breiler** [comment]
I use MacOSX (though an older version) to develop UGS.

Can you please attach the log files:
~/Library/Application Support/ugsplatform/var/log/messages.log

**Reply by dkag123** [reply]
Also, I can see a message that appears just before it crashes... something about 'old setting files'..? It goes away to quick for me to read

**Comment by dkag123** [comment]
[messages.log](https://github.com/winder/Universal-G-Code-Sender/files/11790561/messages.log)

Thanks, I have since installed .13 and .14 versions to try and get around the 'door' message (with no luck) but I i opened .17 to generate the log message, if you need something else let me know

**Reply by breiler** [reply]
The log file did not give any clues. You could be so unlucky that the setting files have been corrupt. 

There are two directories that you could try and delete:
~/Library/Preferences/ugs
~/Library/Application Support/ugsplatform

**Reply by dkag123** [reply]
I uninstalled everything (all versions) and tried again (also deleted the 2 directories you mention above) Still no luck.
After I get the crash I get the OS pop up asking if I want to re-open, If I click to reopen I get a message that Java cant be opened because it is unsigned.. i went to sys preferences and had it 'open anyway'.. Java itself then opens successfully in a terminal window.. but I still cant get the UGS to open (I tried while leaving Java open as well) Here is the log file again without any other versions of the app on the system, just in case 
[messages.log](https://github.com/winder/Universal-G-Code-Sender/files/11790808/messages.log)

**Reply by mdangc** [reply]
I'm facing the same issue here where it kept crashing

**Reply by breiler** [reply]
@mdangc please create a new discussion or  bug report. I want to know your OS version, UGS version and preferably your log file (open menu Tools -> Open log directory)

**Reply by mdangc** [reply]
So I figured it out! It's the Java version which included in the UGS 2.0.14 build for MacOS. I modified the file path to Java written in ugsconf from /etc folder to uses the java version comes with MacOS instead -> (usr/bin). Then the app will launch again!

However, this method only works for UGS 2.0.14 and 2.015. I tested it on the newer build and the crash still persisted.

**Reply by breiler** [reply]
Is there a reason you are using those old (ancient) versions?

Since you still haven't provided details about your OS I will have to guess that it is running the newer M1 or M2 processor. You will need an ARM64 version of Java which we added support for in version 2.1.0. 

Have you tried the latest version 2.1.8?

**Reply by mdangc** [reply]
I tried the latest version 2.1.8 and it’s still crashing. For now, I think 2.0.14 works in my case. 

I can’t provide details about the MacOS neither the hardware because of NDA. It’s probably the OS caused it so I don’t think UGS would be the issue here. Thank you for your help!
